### ISSUE TITLE
Minimum RTL altitude not abided by

### ISSUE THREAD

**Issue by ryanjAA** [PX4/PX4-Autopilot#20644]
On Main (see log - SITL).

When minimum Return altitude is set higher to current alt and the vehicle is put in return, it doesnt climb.

https://review.px4.io/plot_app?log=740ea31d-34e3-4904-888f-1607481cabb2

**Comment by junwoo091400** [comment]
Sort of related to https://github.com/PX4/PX4-Autopilot/pull/19971, @MaEtUgR didn't you say that the RTL altitude setting is practically ignored under certain conditions due to the RTL cone logic?

**Comment by sfuhrer** [comment]
No, not related to https://github.com/PX4/PX4-Autopilot/pull/19971, as the cone is only active in MC mode and the flight here is in FW.

**Comment by sfuhrer** [comment]
@ryanjAA can you reproduce with latest `main`? I quickly tried but couldn't.
